### PR TITLE
Remove moment in favor of date-fns

### PR DESCRIPTION
**Pull request by myarmolinsky**
Remove moment in favor of date-fns

Resolves: #1608 
Change-type: patch
Signed-off-by: Matthew Yarmolinsky <matthew-timothy@balena.io>

---

##### Contributor checklist

- [ ] I have regenerated jest snapshots for any affected components with `npm run generate-snapshots`

##### Reviewer Guidelines

- When submitting a review, please pick:
  - '_Approve_' if this change would be acceptable in the codebase (even if there are minor or cosmetic tweaks that could be improved).
  - '_Request Changes_' if this change would not be acceptable in our codebase (e.g. bugs, changes that will make development harder in future, security/performance issues, etc).
  - '_Comment_' if you don't feel you have enough information to decide either way (e.g. if you have major questions, or you don't understand the context of the change sufficiently to fully review yourself, but want to make a comment)

---

### CODE REVIEW

**Comment by myarmolinsky** [comment]
@balena-ci I self certify!

File: src/unstable-temp/DownloadImageModal/FormModel.tsx [hunk 1]
```diff
@@ -208,9 +208,7 @@ const FormControl = ({ onModelChange, options, model }: FormControlProps) => {
 								name={options.name}
 								value={
 									!!model[options.name]
-										? moment(model[options.name] as string)
-												.utc()
-												.format()
+										? parseISO(model[options.name] as string).toISOString()
```
Comment (iamsolankiamit) [inline src/unstable-temp/DownloadImageModal/FormModel.tsx:211]: would the date returned in `model[options.name]` will always be an ISO string conforming to ISO 8601 format (`YYYY-MM-DDTHH:mm:ss.sssZ`)? If it does then using `new Date()` or `Date.parse()` should also work.  We wouldn't even need `date-fns`

File: src/unstable-temp/DownloadImageModal/ImageForm.tsx [hunk 2]
```diff
@@ -413,11 +413,9 @@ export const ImageForm = ({
 					onClick={(event: React.MouseEvent) => {
 						if (downloadOptions.provisioningKeyExpiryDate) {
 							// Convert to UTC for /download
-							downloadOptions.provisioningKeyExpiryDate = moment(
+							downloadOptions.provisioningKeyExpiryDate = parseISO(
```
Comment (iamsolankiamit) [inline src/unstable-temp/DownloadImageModal/ImageForm.tsx:416]: @myarmolinsky Sorry I didn't mention it here as well. would it also be a valid ISO string?

Comment (myarmolinsky) [reply src/unstable-temp/DownloadImageModal/ImageForm.tsx:416]: @iamsolankiamit It works but we need to convert it to UTC too

Comment (myarmolinsky) [reply src/unstable-temp/DownloadImageModal/ImageForm.tsx:416]: Nevermind it works, idk why it appeared to me that it did not at first ...

Comment (myarmolinsky) [reply src/unstable-temp/DownloadImageModal/ImageForm.tsx:416]: Good catches, thank you 🚀

**Pull request merged**